akka-javasdk/src/main/scala/akka/javasdk/impl/SdkRunner.scala

```diff
@@ -598,7 +597,7 @@ private final class Sdk(
 
         // FIXME pull this inline setup stuff out of SdkRunner and into some workflow class
         val workflowStateType: Class[S] = Reflect.workflowStateType(workflow)
-        messageCodec.registerTypeHints(workflowStateType)
+        serializer.registerTypeHints(workflowStateType)
```

Comment:
@octonato this is of importance for workflows

Reply:
yep, I just rebase and fixed my branch. 

anyway, let's do it.